src/omlt/io/onnx_parser.py

```diff
@@ -0,0 +1,368 @@
+                    node, next_nodes
+                )
+                if new_layer is not None:
+                    network.add_layer(new_layer)
+                    for layer_input in new_layer_inputs:
+                        network.add_edge(layer_input, new_layer)
+            else:
+                for next in next_nodes:
+                    self._node_stack.append(next)
+
+        return network
+
+    def _visit_node(self, node, next_nodes):
+        if node.op_type == "MatMul":
+            next_nodes, new_layer, new_layer_inputs = self._consume_dense_nodes(node, next_nodes)
+        elif node.op_type == "Gemm":
+            next_nodes, new_layer, new_layer_inputs = self._consume_gemm_dense_nodes(node, next_nodes)
+        elif node.op_type == "Conv":
+            next_nodes, new_layer, new_layer_inputs = self._consume_conv_nodes(node, next_nodes)
+        elif node.op_type == "Reshape":
+            next_nodes = self._consume_reshape_nodes(node, next_nodes)
+            new_layer = new_layer_inputs = None
+        else:
+            raise Exception(f"Unhandled node type {node.op_type}")
+
+        for next in next_nodes:
+            self._node_stack.append(next)
+
+        return new_layer, new_layer_inputs
+
+    def _consume_dense_nodes(self, node, next_nodes):
+        """Starting from a MatMul node, consume nodes to form a dense Ax + b node."""
+        assert node.op_type == "MatMul"
+        assert len(node.input) == 2
+
+        [in_0, in_1] = list(node.input)
+        input_layer, transformer = self._node_input_and_transformer(in_0)
+        node_weights = self._initializers[in_1]
+
+        assert len(next_nodes) == 1
+
+        # expect 'Add' node ahead
+        type_, node, maybe_next_nodes = self._nodes[next_nodes[0]]
+        assert type_ == "node"
+        assert node.op_type == "Add"
+
+        # extract biases
+        next_nodes = maybe_next_nodes
+        assert len(node.input) == 2
+        [in_0, in_1] = list(node.input)
+
+        if in_0 in self._initializers:
+            node_biases = self._initializers[in_0]
+        else:
+            assert in_1 in self._initializers
+            node_biases = self._initializers[in_1]
+
+        assert len(node_weights.shape) == 2
+        assert node_weights.shape[1] == node_biases.shape[0]
+        assert len(node.output) == 1
+
+        input_output_size = input_layer.output_size
+        if transformer is not None:
+            input_output_size = transformer.output_size
+
+        output_size = input_output_size[:-1] + [node_weights.shape[1]]
+
+        activation = "linear"
+        if len(next_nodes) == 1:
+            # check if Relu
+            type_, maybe_node, maybe_next_nodes = self._nodes[next_nodes[0]]
+            if maybe_node.op_type in _ACTIVATION_OP_TYPES:
+                node = maybe_node
+                activation = maybe_node.op_type.lower()
+                next_nodes = maybe_next_nodes
```

Comment:
will this use a linear activation if `maybe_node.op_type` is not in `_ACTIVATION_OP_TYPES`? this could lead to an incorrect neural network

Reply:
this happens in a few spots in the onnx parser. we should consider checking for un-supported activation types

Reply:
I made this an issue. We need to check both onnx_parser and keras_reader